Bozza prima sezione e struttura (#162)
Il-Libro-Open-Source/book: docs/it/come-affrontare-l-eta-nel-mondo-it.md

- Bisogna per forza diventare manager dopo X anni? (Posso continuare ad essere dev, oppure devo diventare manager?) Quali sono i percorsi alternativi che permettono di essere sempre in prima linea sull'aspetto tecnico? E in questi percorsi, si scriverà ancora codice.
- È possibile essere competitivi sul mercato come lo sarebbe una figura più giovane? Come si comportano le aziende sulla questione?
- Esempi reali (se possibile) di come molte volte l'esperienza faccia la differenza rispetto a conoscere bene le tecnologie all'ultimo grido.
- Quali obiettivi/consapevolezze/conoscenze bisogna raggiungere dopo X anni di esperienza in questo campo? (Molto spesso gli anni di esperienza non valgono nulla se per tanti anni si fanno le stesse cose).

# Introduzione

In questo capitolo affronteremo il tema dell'età nel mondo IT, un tema che spesso viene sottovalutato ma che è molto importante per chi lavora in questo settore e che spesso si trova a dover fare i conti con la propria età chiedendosi se è ancora competitivo sul mercato del lavoro, se è ancora in grado di stare al passo con le nuove tecnologie o se invece è costretto a districarsi tra ruoli manageriali e non più tecnici.

> E' opportuno precisare che l'obiettivo di questo capitolo non è di certo dare delle certezze assolute o percorsi di carriera uguali per tutti, in quanto ogni persona è unica e diversa in termini di passioni, ambizioni e attitudini. Faremo comunque delle considerazioni che potranno aiutarti a compredere come vuoi vivere la tua carriera dopo molti anni di esperienza e riflettere sulle implicazioni che potrebbero comportare alcune scelte.

Reviewer: ```suggestion
E' opportuno precisare che l'obiettivo di questo capitolo non è di certo dare delle certezze assolute o percorsi di carriera uguali per tutti, in quanto ogni persona è unica e diversa in termini di passioni, ambizioni e attitudini. Faremo infatti delle considerazioni personali che potranno aiutarti a compredere come vuoi vivere la tua carriera dopo molti anni di esperienza e riflettere sulle implicazioni che potrebbero comportare alcune scelte.
```

# Posso continuare ad essere dev, oppure devo diventare manager?

> Molto probabilmente questa è una domanda che ti sei già posto qualche volta, oppure ti sarà capitato di fare considerazioni simili coi tuoi colleghi durante la pause pranzo, o, addiruttura anche col tuo referente nei momenti dedicati alle valutazioni delle tue perfomance. Possiamo affermare che la maggior parte dei devs, con già qualche anno di esperinza alle spalle, risponderebbe che è possibile rimanere dev anche dopo X anni di esperienza e continuare senza dover abbandonare il codice. D'altronde se si è diventati developer per pura passione dello sviluppo, perché bisogna smettere?

Reviewer: ```suggestion
Molto probabilmente questa è una domanda che ti sei già posto qualche volta, oppure ti sarà capitato di fare considerazioni simili con i tuoi colleghi durante la pause pranzo, o, addirittura anche con il/la tuo/a referente nei momenti dedicati alle valutazioni delle tue performance. Possiamo affermare che la maggior parte di chi sviluppa, con già qualche anno di esperienza alle spalle, risponderebbe che è possibile rimanere dev anche dopo decine di anni di esperienza e continuare senza dover abbandonare il codice. D'altronde se si è diventati developer per pura passione dello sviluppo, perché smettere?
```

> Forse, caro lettore, penserai che con l'età che inizia ad avanzare può risultare più arduo rimanere al passo con le tecnologie e che di conseguenza è difficile rimanere competitivi rispetto ad una persona più giovane, forse più aggiornata e stimolata. Pensiero assolutamente corretto e condivisibile, considerando che ragionamenti analoghi possono essere fatti anche su discipline diverse dal mondo IT. Ci sono articoli su articoli che puoi trovare in rete nei quali si evidenzia su come sia più complicato in età adulta trovare lavoro, con consigli su come reiventarsi o su come rimanere competitivi con il mondo del lavoro che continua ad evolversi di anno in anno.

Reviewer: ```suggestion
Forse, caro lettore, penserai che con l'età che inizia ad avanzare possa risultare più arduo rimanere al passo con le tecnologie e che di conseguenza possa essere difficile rimanere competitivi rispetto ad una persona più giovane, spesso più aggiornata e stimolata. Pensiero assolutamente corretto e condivisibile, considerando che ragionamenti analoghi possono essere fatti anche su discipline diverse dal mondo IT. Ci sono articoli su articoli che puoi trovare in rete nei quali si evidenzia come sia più complicato in età adulta trovare lavoro, con consigli su come re-inventarsi o su come rimanere competitivi in un mondo del lavoro che continua ad evolversi di anno in anno.
```

> Ma torniamo al pianeta IT: se hai già letto il capitolo _Cosa significa "essere dev"?_ avrai già realizzato che essere dev è molto cose, non solo linguaggi, librerie e frameworks. Chi fa sviluppo deve realizzare che questi strumenti servono per uno scopo: risolvere problemi per aiutare il business. Conoscere la nuova tecnologia non aiuta molto il business di cui sei parte se non si applica bene al tuo dominio o necessità del momento. Molto spesso un dev giovane con tanta voglia di fare e sperimentare, cade in questo errore. Una persona che sviluppa da tanti anni potrà dare un contributo di valore nell'ambiente in cui si trova e aiutare anche i devs più giovani a fare scelte più ponderate e corrette. Tu che leggi, che forse ancora non hai deciso come vuoi impostare la tua carriera, al di là della tua età e degli anni di esperienza, se vuoi fare appieno questo lavoro, poniti come obiettivo di non scrivere solo "bel codice", ma di portare _valore_ in ciò che fai per la tua azienda.

Reviewer: ```suggestion
Ma torniamo al pianeta IT: se hai già letto il capitolo _Cosa significa "essere dev"?_ avrai già realizzato che essere dev è molto cose, non solo linguaggi, librerie e frameworks. Chi fa sviluppo deve realizzare che questi strumenti servono per uno scopo: risolvere problemi per aiutare il business. Conoscere la nuova tecnologia non aiuta molto il business di cui sei parte se non si applica bene al tuo dominio o necessità del momento. Molto spesso un dev giovane con tanta voglia di fare e sperimentare, cade in questo errore. Una persona che sviluppa da tanti anni potrà dare un contributo di valore nell'ambiente in cui si trova e aiutare anche dev più giovani a fare scelte più ponderate e corrette. Tu che leggi, che forse ancora non hai deciso come vuoi impostare la tua carriera, al di là della tua età e degli anni di esperienza, se vuoi fare appieno questo lavoro, poniti come obiettivo di non scrivere solo "bel codice", ma di portare _valore_ in ciò che fai per la tua azienda.
```

## Pensare al "vero" obiettivo

Ripetiamo: il lavoro del developer è (anche) risolvere problemi. Il nostro scopo non è, e non sarà sempre, trovare soluzioni accativanti, magari con le tecnologie del momento. Ed è questa la maturità che deve acquisire un developer negli anni e con l'esperienza.

> Vogliamo evidenziare questo aspetto con un aneddoto riportato nel libro Clean Architecure di Robert C. Martin. Senza entrare troppo nei dettagli che puoi trovare nel libro, durante una sua esperienza lavorativa alla fine degli anni Ottanta, ha dovuto, per pura esigenza di marketing, impiegare un database relazionale in sistema di cui occupava. Ma per motivi ingegneristici lui affermava che non era assolutamente necessario avere un database in quel sistema, che raggiungeva appieno il suo scopo così com'era, ma il mercato di quel perido voleva i databasa nei sistemi, e l'azienda "accontentò" il mercato. Dal canto suo, Martin era focalizzato sull'obiettivo igegneristoco reale, e crediamo che debba essere questa una compentenza di un developer.

Reviewer: ```suggestion
Vogliamo evidenziare questo aspetto con un aneddoto riportato nel libro Clean Architecure di Robert C. Martin. Senza entrare troppo nei dettagli che puoi trovare nel libro, durante una sua esperienza lavorativa alla fine degli anni Ottanta, ha dovuto, per pura esigenza di marketing, impiegare un database relazionale in un sistema di cui occupava. Ma per motivi ingegneristici lui affermava che non era assolutamente necessario avere un database in quel sistema, che raggiungeva appieno il suo scopo così com'era, ma il mercato di quel periodo voleva i database nei sistemi, e l'azienda _accontentò_ il mercato. Dal canto suo, Martin era focalizzato sull'obiettivo ingegneristico reale, e crediamo che debba essere questa la competenza che chi fa questo lavoro dovrebbe sviluppare col tempo.
```

Facciamo un esempio più recente: i microservizi. Chissà in quanti annunci di lavoro li avrai visti in questi ultimi anni.

> Al di là dei vantaggi (e svantaggi) di questa architettura, sembra che non se ne possa più fare a meno, la panacea di tutti i mali. Ma nel marzo 2023, il team di Prime Video hanno pubblicato un post sui loro blog, che col passaggio da un'architettura distruibita di microservizi a un'architettura monolitica ha ridotto costi infrastrutturali di oltre il 90%. Il team di Prime Video, si è focalizzato su obiettivo preciso: ottimizzare i costi. E ci sono riuscito con il tanto "odiato" monolite.

Reviewer: ```suggestion
Al di là dei vantaggi (e svantaggi) di questa architettura, sembra che non se ne possa più fare a meno, la panacea di tutti i mali. Ma nel marzo 2023, il team di Prime Video ha pubblicato un post sul proprio blog, indicando che con il passaggio da un'architettura distribuita a microservizi a un'architettura monolitica ha ridotto i costi infrastrutturali di oltre il 90%. Il team di Prime Video si è focalizzato su un obiettivo preciso: ottimizzare i costi. E ci sono riusciti con una soluzione tanto semplice quanto _classica_, il monolite.
```

> I due esempi, sono diversi in termini di situazione e tempi, ma ci fanno comprendere una cosa: il nostro lavoro è cercare il modo migliore di raggiungere l'obiettivo ingegneristico che aiuta al business, e non le mode del momento.

Reviewer: ```suggestion
I due esempi sono diversi in termini di situazione e tempi, ma ci fanno comprendere una cosa: il nostro lavoro è cercare il modo migliore di raggiungere l'obiettivo ingegneristico che aiuti al business, a prescindere dalle mode del momento.
```

> Se sei un developer che sa cosa da valore nelle scelte che dobbiamo fare, perché preoccuparsi se l'età avanza?

Reviewer: ```suggestion
Se sei un developer che sa cosa dia valore nelle scelte che dobbiamo fare, perché preoccuparsi se l'età avanza?
```

## Diventare manager

> Facciamo una breve considerazione sul diventare manager in modo da giungere alle dovute riflessioni che potranno esserti utile quando un giorno dovrai fare questa scelta. Per manager intendiamo, per semplicità e senza andare troppo nel dettaglio, quei ruoli che non sono coinvolti nelle decisoni e operativi per quanto concerne l'aspetto tecnico, ma sostanzialmente incentrati nella gestione di alto livello dei progetti e dei team.

Reviewer: ```suggestion
Facciamo una breve considerazione sul diventare manager in modo da giungere alle dovute riflessioni che potranno esserti utili quando un giorno dovrai fare questa scelta. Per manager intendiamo, per semplicità e senza andare troppo nel dettaglio, quei ruoli che non sono coinvolti nelle decisioni e operativi per quanto concerne l'aspetto tecnico, ma sostanzialmente incentrati nella gestione di alto livello dei progetti e dei team.
```

> Abbiamo constatato che è possibile rimanere dev anche con l'avanzare dell'età, ma molto spesso diventare manager sembra quasi l'unico modo per fare carriera. Ci sono realtà aziendali che hanno una gerarchia tale per cui chi fa sviluppo sembra essere giunto ad un primo step della carriera, una specie di base dalla quale poter avanzare, prima o poi, abbandonando la via del codice. Noi ci auguriamo che questo non sia il tuo caso, ma può capiare che il contesto lavorativo in cui ci si trova influenzi il modo di percepire la carriera e di conseguenza le nostre scelte, perché vediamo un'unica strada davanti a noi. Questo potrebbe far pensare che non raggiungere un ruolo manageriale dopo molti anni di esperienza equivalga a un fallimento nella tua carriera, sentendoti inadeguato sul mercato. Per evitare ciò, si diventa manager per raggiungere una sorta di traguardo professionale e, perchè no, anche salariale.

Reviewer: ```suggestion
Abbiamo constatato che è possibile rimanere dev anche con l'avanzare dell'età, ma molto spesso diventare manager sembra quasi l'unico modo per fare carriera. Ci sono realtà aziendali con una gerarchia tale per cui chi fa sviluppo è giunto solo ad un primo step della propria carriera, una base dalla quale poter avanzare, prima o poi, abbandonando la via del codice, verso quella del manager. Noi ci auguriamo che questo non sia il tuo caso, ma può capitare che il contesto lavorativo in cui ci si trova influenzi il modo di percepire la carriera e di conseguenza le nostre scelte, perché vediamo un'unica strada davanti a noi. Questo potrebbe far pensare che non raggiungere un ruolo manageriale dopo molti anni di esperienza equivalga a un fallimento nella tua carriera, sentendoti inadeguato sul mercato. Per evitare ciò, si diventa manager per raggiungere una sorta di traguardo professionale e, perchè no, anche salariale.
```

> Caro lettore, vogliamo dirti che non è così: se in te c'è passione per lo sviluppo non sei costretto a intrapendere percorsi manageriali. La programmazione offre davvero tane sfide e soddisfazioni che puoi trovare in contesti diversi in cui forse ti trovi attualmente. Continua a studiare, ad approfondire e confrontati con altre persone con la nostra stessa passione dove puoi scoprire che c'è un mondo del lavoro per i developers molto stimolante. E se realizzi che ti trovi nel contesto non adatto a te, cerca realtà aziendali dove ci sono percorsi alternativi, che ti permetteranno di crescere tecnicamente e di continuare a crescere.

Reviewer: ```suggestion
Caro lettore, vogliamo dirti che non è così: se in te c'è passione per lo sviluppo non sei costretto a intraprendere percorsi manageriali. La programmazione offre molte sfide e soddisfazioni che puoi trovare in contesti diversi in quello nel quale probabilmente ti trovi attualmente. Continua a studiare, ad approfondire e confrontarti con altre persone con la tua stessa passione. E se dovessi realizzare di non trovarti nel contesto non adatto a te, cerca realtà aziendali dove ci sono percorsi alternativi, che ti permetteranno di crescere tecnicamente e di continuare a perseguire i tuoi obiettivi.
```

> Concludiamo questa sezione dicendoti che non vogliamo convincerti a non diventare manager. Ci sono tante sfide in questo ruolo che sicuramente dà molte soddisfazioni. Diventa manager, abbandonando pure la parte tecnica e il codice, se ti rivedi in questa figura, se questo ruolo ti appassiona e sei convinto di questa decisione. Siamo certi che se intraprendi questo percorso con decisione e passione potrai diventare un manager con il quale tutti i devs avranno piacere di collaborare.

Reviewer: ```suggestion
Concludiamo questa sezione dicendoti che non vogliamo convincerti a non diventare manager. Ci sono tante sfide in questo ruolo che sicuramente ti darebbero molte soddisfazioni. Diventa manager, abbandonando pure la parte tecnica e il codice, se ti rivedi in questa figura, se questo ruolo ti appassiona e sei convinto di questa decisione. Siamo certi che se intraprenderai questo percorso con decisione e passione potrai diventare un manager con il quale tutti avranno piacere di collaborare.
```

Caro lettore, vogliamo dirti che non è così: se in te c'è passione per lo sviluppo non sei costretto a intraprendere percorsi manageriali. La programmazione offre molte sfide e soddisfazioni che puoi trovare in contesti diversi in quello nel quale probabilmente ti trovi attualmente. Continua a studiare, ad approfondire e confrontarti con altre persone con la tua stessa passione. E se dovessi realizzare di non trovarti nel contesto non adatto a te, cerca realtà aziendali dove ci sono percorsi alternativi, che ti permetteranno di crescere tecnicamente e di continuare a perseguire i tuoi obiettivi.

Concludiamo questa sezione dicendoti che non vogliamo convincerti a non diventare manager. Ci sono tante sfide in questo ruolo che sicuramente ti darebbero molte soddisfazioni. Diventa manager, abbandonando pure la parte tecnica e il codice, se ti rivedi in questa figura, se questo ruolo ti appassiona e sei convinto di questa decisione. Siamo certi che se intraprenderai questo percorso con decisione e passione potrai diventare un manager con il quale tutti avranno piacere di collaborare.

## Non _solo_ developers

> In questa sezione rifletteremo brevemente su una domanda in apparenza semplice ma allo stesso tempo interessante: se desideriamo continuare a scrivere codice, è necessario rimanere developers per sempre?

Reviewer: ```suggestion
In questa sezione rifletteremo brevemente su una domanda in apparenza semplice ma allo stesso tempo interessante: se desideriamo continuare a scrivere codice, è necessario rimanere developer per sempre?
```

Probabilmente nella tua mente starai già pensando al tuo contesto aziendale o alle tue precedenti esperienze, cercando figure _ibride_ tra ruoli tecnici e manageriali: per citarne alcune ci sono figure come Team Leader, Tech Lead e Solution Architect a cui i developers fanno riferimento per ciò che riguarda decisioni più o meno delicate su aspetti tecnici. Come forse saprai, non è detto che queste figure, prese da altre responsabilità o coinvolte in altri processi più organizzativi, riescano ancora effettivamente a scrivere codice. Ogni realtà solitamente ridefinisce questi ruoli, proiettandoli nei propri contesti. Per esempio, un team leder in una start-up che è poco strutturata è probabile che continui a sviluppare, rispetto a realtà più grandi e strutturate dove il team leader è più simile ad una figura manageriale. Ci sono, invece, realtà molto flessibili che danno la libertà ai propri team leader di gestirsi e decidere quanto addentrarsi negli aspetti tecnici.

> Questi ruoli sono molto interessanti e sfidanti qualora fare _solo coding_ iniziasse a non bastarti. Se vuoi gestire un team e avere responsabiltà più importanti, sicuramente prima o poi potrai intrapendere uno di questi percorsi senza rinunciare del tutto all'aspetto tecnico.

Reviewer: ```suggestion
Questi ruoli sono molto interessanti e sfidanti qualora fare _solo coding_ iniziasse a non bastarti. Se vuoi gestire un team e avere responsabilità differenti, sicuramente prima o poi potrai intraprendere uno di questi percorsi senza rinunciare del tutto all'aspetto tecnico.
```

Preferisco usare il termine "responsabilità differenti", non credo che un manager ne abbia di più "importanti", se il software ha dei bug questa cosa è impattante almeno quanto una pessima decisione manageriale :)

Author: Condivdo appieno!

> Al di là della tua scelta, ricorda che se vuoi essere un developer che aggiunge valore nonostante l'età, non è sufficiente che ti limiti al coding, cerca sempre di contribuire a ogni aspetto del progetto: sia che tu rivesta il ruolo di developer o di team leader, impegnati costantemente nel risolvere problemi e suggerire soluzioni ottimali, senza concentrarti sui limiti assegnati dai ruoli.

Reviewer: ```suggestion
Al di là della tua scelta, ricorda che se vuoi essere un developer che aggiunga valore nonostante l'età, non è sufficiente che ti limiti al coding, cerca sempre di contribuire a ogni aspetto del progetto: sia che tu rivesta il ruolo di developer o di team leader, impegnati costantemente nel risolvere problemi e suggerire soluzioni ottimali, senza concentrarti sui limiti assegnati dai ruoli.
```

> Se sei ancora un junior, proponi soluzioni anche se non sei del tutto sicuro della loro correttezza. Non solo l'impegno nello studio e nell'approfondimento, ma anche trarre insegnamento dai propri errori rappresenta uno dei migliori modi per crescere.

Reviewer: ```suggestion
Se sei ancora junior, proponi soluzioni anche se non sei del tutto sicuro della loro correttezza. Non solo l'impegno nello studio e nell'approfondimento, ma anche e soprattutto trarre insegnamento dai propri errori è un ottimo modo per crscere.
```

Così, dopo anni di esperienza, potrai mettere in campo molte conoscenze che figure più giovani del team probabilmente non avranno.

Un developer che sa comunicare e sa trasmettere le proprie conoscenze ad altri, sarà sempre di grande valore per un team e avrà un'autorevolezza al suo interno a prescindere dal ruolo e soprattutto dall'età.

## Conclusioni

> In questo capitolo abbiamo parlato di come affrontare l'età nel mondo IT. Questo è sempre un tema delicato, e non riguarda solo il mondo IT, senza considerare che ci sono tanti fattori collegati ad esso: famiglia, figli, tempo che non basta mai per aggiornarsi e studiare, stanchezza ecc..

Reviewer: ```suggestion
In questo capitolo abbiamo parlato di come affrontare l'età nel mondo IT. Questo è sempre un tema delicato, e non riguarda solo il mondo IT, senza considerare che ci sono tanti fattori collegati ad esso: famiglia, figli, tempo che non basta mai per aggiornarsi e studiare, stanchezza, ecc..
```

> Ma il nostro obiettivo era di darti modo di fare le tue considerazioni, con l'auspicio che ciò possa portarti a una maggiore chiarezza riguardo al tuo percorso professionale futuro, e che la passione, la volontà di dare sempre il tuo contributo e di condividere le tue esperienze con il team, possano continuare a farti a sviluppare con entusiasmo senza timore dell'età che avanza.

Reviewer: ```suggestion
L'obiettivo era di darti modo di fare le tue considerazioni, con l'auspicio che ciò possa portarti ad una maggiore chiarezza riguardo al tuo percorso professionale futuro, e che la passione, la volontà di dare sempre il tuo contributo e di condividere le tue esperienze con il team, possano continuare a farti a sviluppare con entusiasmo senza timore dell'età che avanza.
```

Concludiamo questa sezione dicendoti che non vogliamo convincerti a non diventare manager. Ci sono tante sfide in questo ruolo che sicuramente ti darebbero molte soddisfazioni. Diventa manager, abbandonando pure la parte tecnica e il codice, se ti rivedi in questa figura, se questo ruolo ti appassiona e sei convinto di questa decisione. Siamo certi che se intraprenderai questo percorso con decisione e passione potrai diventare un manager con il quale tutti avranno piacere di collaborare.

> ## Non _solo_ developers

Reviewer: ```suggestion
## Non _solo_ developer
```

Reviewer: La s finale tende a pluralizzare in inglese, ma essendo il contenuto in italiano è da rimuovere, vero @Il-Libro-Open-Source/drafting-group ?

Author: L'osservazione mi torna.
Attendiamo altri pareri per conferma.
